libsquoosh/src/index.ts

```diff
@@ -161,14 +195,20 @@ class Image {
   /**
    * Define one or several encoders to use on the image.
    * @param {object} encodeOptions - An object with encoders to use, and their settings.
-   * @returns {Promise<undefined>} - A promise that resolves when the image has been encoded with all the specified encoders.
+   * @returns {Promise<void>} - A promise that resolves when the image has been encoded with all the specified encoders.
    */
-  async encode(encodeOptions = {}) {
+  async encode(
+    encodeOptions: {
```

Comment:
I'm not sure if this is the correct typing for the `encode` method. For example, this is how it is called in the example in readme:
```
const encodeOptions = {
  mozjpeg: {}, //an empty object means 'use default settings'
  jxl: {
    quality: 90,
  },
}
await image.encode(encodeOptions);
```

It should be something like:
```
{
optimizerButteraugliTarget?: number;
maxOptimizerRounds?: number;
} & {
    [key: EncoderKey]: any; // any is okay for now
}
```

I'm not sure about the `[key: EncoderKey]` syntax though 😅

Reply:
It looks like the correct syntax is `[key in EncoderKey]?: any; // any is okay for now`. WDYT?